manifests/minio/minio.yaml

```diff
@@ -0,0 +1,135 @@
+# NOTE:
+# It is necessary to first deploy a secret that minio will use
+# for retrieving the username and password. To generate it
+# execute:
+# $ kubectl create secret generic minio-key --namespace kubeless --from-literal=accesskey=<username> --from-literal=secretkey=<password>
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: minio
+  namespace: kubeless
+  annotations:
+    volume.alpha.kubernetes.io/storage-class: standard
+spec:
+  storageClassName: standard
+  accessModes:
+    - "ReadWriteOnce"
+  resources:
+    requests:
+      storage: "10Gi"
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: minio
+  namespace: kubeless
+  labels:
+    kubeless: minio
+spec:
+  selector:
+    kubeless: minio
+  ports:
+    - port: 9000
+      targetPort: 9000
+      protocol: TCP
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
```

Comment:
A Deployment with a PVC has some odd behaviours during update/failure recovery, while the replicas fight over access to the PV.  I suggest you change this to a StatefulSet, or modify the updateStrategy appropriately.